gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetTest.java

```diff
@@ -176,22 +176,72 @@ public void testGenerateCopyEntitiesMultiSnapshotWhenDestEmpty() throws IOExcept
+
+    MockFileSystemBuilder destBuilder = new MockFileSystemBuilder(DEST_FS_URI);
+    FileSystem destFs = destBuilder.build();
+
+    CopyConfiguration copyConfiguration = CopyConfiguration.builder(destFs, copyConfigProperties)
+        .preserve(PreserveAttributes.fromMnemonicString("ugp"))
+        .copyContext(new CopyContext())
+        .build();
+    Collection<CopyEntity> copyEntities = icebergDataset.generateCopyEntities(destFs, copyConfiguration);
+    verifyFsOwnershipAndPermissionPreservation(copyEntities);
+  }
+
   private void verifyCopyEntities(Collection<CopyEntity> copyEntities, List<String> expected) {
     List<String> actual = new ArrayList<>();
     for (CopyEntity copyEntity : copyEntities) {
       String json = copyEntity.toString();
-      String filepath = new Gson().fromJson(json, JsonObject.class)
-          .getAsJsonObject("object-data").getAsJsonObject("origin")
-          .getAsJsonObject("object-data").getAsJsonObject("path")
-          .getAsJsonObject("object-data").getAsJsonObject("uri")
-          .getAsJsonPrimitive("object-data").getAsString();
+      String filepath = getFilePathAsStringFromJson(json);
       actual.add(filepath);
     }
     Assert.assertEquals(actual.size(), expected.size(),
         "Set" + actual.toString() + " vs Set" + expected.toString());
     Assert.assertEqualsNoOrder(actual.toArray(), expected.toArray());
   }
 
+  private void verifyFsOwnershipAndPermissionPreservation(Collection<CopyEntity> copyEntities) {
+    List<JsonObject> ancestorsOwnerAndPermissionsList = new ArrayList<>();
+    List<JsonObject> destinationOwnerAndPermissionsList = new ArrayList<>();
+
+    for(CopyEntity copyEntity : copyEntities) {
+      String copyEntityJson = copyEntity.toString();
+
+      JsonArray ancestorsOwnerAndPermissions = new Gson().fromJson(copyEntityJson, JsonObject.class)
+          .getAsJsonObject("object-data").getAsJsonArray("ancestorsOwnerAndPermission");
+      JsonObject rootAncestorOwnerAndPermissions = ancestorsOwnerAndPermissions.get(ancestorsOwnerAndPermissions.size()-1)
+          .getAsJsonObject().getAsJsonObject();
+      ancestorsOwnerAndPermissionsList.add(rootAncestorOwnerAndPermissions);
+      String filePath = getFilePathAsStringFromJson(copyEntityJson);
+      Assert.assertEquals(ancestorsOwnerAndPermissions.size(), new Path(filePath).getParent().depth() -1);
+
+      JsonObject destinationOwnerAndPermissions = new Gson().fromJson(copyEntityJson, JsonObject.class)
+          .getAsJsonObject("object-data").getAsJsonObject("destinationOwnerAndPermission");
+      destinationOwnerAndPermissionsList.add(destinationOwnerAndPermissions);
+    }
+    Assert.assertEqualsNoOrder(ancestorsOwnerAndPermissionsList.toArray(), destinationOwnerAndPermissionsList.toArray());
+  }
+
+  private String getFilePathAsStringFromJson(String json) {
+    String filepath = new Gson().fromJson(json, JsonObject.class)
```

Comment:
first off, I *love* that you've raised the level of abstraction!  that said, is this the right approach of a single helper method?  I see that in the case of verifying permission preservation, you need to initialize a second and even third:
```
Gson().fromJson(copyEntityJson, JsonObject.class)
          .getAsJsonObject("object-data")
```
(which unfortunately go without any encapsulating abstraction).

why not a variation on the Proxy pattern, where you initialize once, hold onto it, and provide getters/extractors?  thus a class, perhaps named, `CopyEntityJsonExtractor`, with methods `extractFilePath`/`extractFileUri` and `extractAncestorsO&P` and `extractDestinationO&P`.

or perhaps clearer naming: `CopyEntityDeserialization` with methods `get*`